lib/src/workspace.rs

```diff
@@ -119,7 +125,17 @@ pub struct Workspace {
 fn create_jj_dir(workspace_root: &Path) -> Result<PathBuf, WorkspaceInitError> {
     let jj_dir = workspace_root.join(".jj");
     match std::fs::create_dir(&jj_dir).context(&jj_dir) {
-        Ok(()) => Ok(jj_dir),
+        Ok(()) => {
+            #[cfg(windows)]
+            #[allow(unsafe_code)]
+            {
+                let jj_dir = CString::new(jj_dir.to_str().unwrap()).unwrap();
```

Comment:
`jj_dir.to_str().unwrap()` will panic if `jj_dir` is not valid utf-8, which I think is not guaranteed here. I think `CString::new(jj_dir.as_os_str().as_encoded_bytes())` should be better.

(The second `.unwrap()` will panic if the string has internal NUL bytes, but I think that is a safe assumption.)